content/en/blog/2025/ux-research-prometheus-otel/index.md

```diff
@@ -0,0 +1,147 @@
+
+One of the most surprising was the realization that [OpenTelemetry’s Resource Detection feature](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/resource#telescoping) allows users to selectively retain or drop resource attributes based on relevance using a conceptual pattern sometimes referred to as “telescoping.” Despite its potential, many users and even some members of the Prometheus community seem unaware of it. This lack of awareness may have contributed to the adoption of the "join" pattern, which has since proven to be problematic.
+
+This highlights a broader issue: documentation and education gaps are a major barrier. In our survey, 78% of respondents cited documentation gaps as a challenge.
+
+Another key realization is that earlier integration decisions, like the reliance on joins, were made without a full understanding of each tool’s capabilities, an unavoidable consequence of the lack of early collaboration and communication between the Prometheus and OpenTelemetry communities.
+
+## Recommended solutions
+
+Based on conversations with stakeholders and end users, here are some of the proposed solutions, grouped by what's feasible in the short term vs. what's part of a longer-term vision:
+
+### Short-term solutions
+
+- **Improved documentation for attribute handling**  
+Since users find it easier to promote attributes than to join on target info, it may be worth de-emphasizing (or even sunsetting) documentation around joins, while making [attribute promotion docs](https://prometheus.io/docs/guides/opentelemetry/#promoting-resource-attributes) more prominent for those not yet aware of the option. The telescoping pattern of resource detection in OpenTelemetry also deserves more visibility and proper documentation. Additionally, users have suggested creating consolidated Prometheus–OpenTelemetry documentation that clearly explains how both systems handle resource attributes.
+
+### Long-term vision
+
+- **Entity framework**  
+OpenTelemetry’s developing concept of entities could help Prometheus distinguish between identifying vs. descriptive attributes. This would guide which attributes become labels, and which are stored or filtered out.
+
+- **Metadata storage**  
+Stakeholders also discussed the idea of adding first-class metadata support to Prometheus itself. This would allow certain resource attributes to be stored as metadata (not labels), avoiding cardinality costs while keeping the information available for queries or joins.
+
+- **Expanding for exploratory telemetry**  
+This might be a stretch, but Prometheus could consider expanding its scope to better support exploratory telemetry use cases. Stakeholders showed openness to change, as long as Prometheus’s core architecture remains intact and existing users aren’t alienated. That suggests there may be room for evolution, especially if new capabilities can complement, rather than replace, current behavior.
+
```

Comment:
Could we also add the quote from Juraci mentioning that Otel was designed for exploratory use cases? I think he says something like this while comparing to Prometheus's design for infrastructure monitoring

Reply:
exactly